src/main/scala/mastermind/Mastermind.scala

```diff
@@ -13,19 +13,30 @@ val validColors = List('B', 'Y', 'R', 'G')
 
 /** Get a random color from the list of valid colors */
 def getRandomColor(): Color =
-  ???
+  val randomIndex = new Random().between(0, 3)
+  validColors.apply(randomIndex)
 
 /** Given four colors, make a board from them */
 def makeBoardFromColors(c1: Color, c2: Color, c3: Color, c4: Color): Board =
-  ???
+  List(c1, c2, c3, c4).mkString
 
 /** Create a random board */
 def getRandomBoard(): Board =
-  ???
+  makeBoardFromColors(
+    getRandomColor(), 
+    getRandomColor(), 
+    getRandomColor(), 
+    getRandomColor())
+
```

Comment:
Your solutions for getRandomBoard and makeBoardFromColors are much cleaner than mine - I made variables, but this is much simpler.